src/chapters/loops/while-loops.rst

```diff
@@ -0,0 +1,199 @@
+
+You can almost read the ``while`` statement as if it were in a natural language: *while ``i`` is less than or equal to ``51``, continue executing the body of the loop*.
+
+Beyond **for** Loops
+--------------------
+
+We stated earlier that ``while`` loops are more flexible than ``for`` loops. Now we will look at an example that illustrates this.
+
+This program is an example of **input validation**. It prompts the user to enter a positive number, converting the input string to the number data type. If the number is not positive, then the user is prompted again within the body of the loop. As long as the user continues to input non-positive numbers, the loop will continue to iterate.
+
+.. sourcecode:: js
+
+   const input = require('readline-sync');
+
+   let num = input.question('Please enter a positive number:');
+   num = Number(num);
+
+   while (num <= 0) {
+      num = input.question('Invalid input. Please enter a positive number:');
+      num = Number(num);
+   }
+
+.. index::
+   pair: input; validation
+
+This example illustrates the additional flexibility provided by ``while`` loops. While we use ``for`` loops to iterate over fixed collections (a string, an array, a collection of integers), the ``while`` loop can be used to iterate in more general circumstances. For the input validation example, at runtime it cannot be determined how many times the loop will repeat.
+
+.. warning:: It is *technically* possible to write a ``while`` loop as a ``for`` loop, since ``for`` loops use a condition, and the initial and update expressions can be left empty. Doing so, however, results in confusing code. This should not be done under any circumstances!
+
+   .. sourcecode:: js
+
+      const input = require('readline-sync');
+
+      let num = input.question('Please enter a positive number:');
+      num = Number(num);
+
+      for (; num <= 0; ) {
+         num = input.question('Invalid input. Please enter a positive number:');
+         num = Number(num);
+      }
+
+Infinite Loops, Revisited
+-------------------------
+
+.. index::
+   single: loop; infinite
+
+It is easier to create an infinite ``while`` loop than an infinite ``for`` loop. To see this, consider our first ``while`` loop example, but where we forget to update the loop variable.
+
+.. sourcecode:: js
+
+   let i = 0;
+
+   while (i < 51) {
+      console.log(i);
+   }
+
+This is an infinite loop. The variable ``i`` is initialized to 0 never updated, so the condition ``i < 51`` will always be true.
+
+Terminating a Loop With **break**
```

Comment:
Suggestion: Shift “Terminating a Loop” to a different rst file.  Since the idea applies for both ``for`` and ``while`` loops, it should not appear in the ``while`` specific file.

Reply:
Good call!